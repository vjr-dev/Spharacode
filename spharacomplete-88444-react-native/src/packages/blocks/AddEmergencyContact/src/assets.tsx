// Customizable Area Start
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const image_back = require("../assets/image_back.png");
export const Rectangle1 = require("../assets/Rectangle1.png");
export const Profile = require("../assets/profile.png");
// Customizable Area End