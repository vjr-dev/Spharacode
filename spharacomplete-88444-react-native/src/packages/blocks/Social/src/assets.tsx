// Customizable Area Start
export const hash = require("../assets/hash.png");
export const chat = require("../assets/chat.png");
export const meeting = require("../assets/meeting.png");
export const more = require("../assets/more.png");
// Customizable Area End