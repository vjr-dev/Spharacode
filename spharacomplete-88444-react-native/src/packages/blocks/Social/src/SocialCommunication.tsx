// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React, { Component } from "react";
import {
  Image,
  ImageSourcePropType,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import {
  createBottomTabNavigator, createStackNavigator
 } from "react-navigation";
import Channels from "../../ChannelsInSocial/src/Channels";
import Chat from "../../ChatInSocial/src/Chat";
import ConversationScreen from "../../ChatInSocial/src/ConversationScreen";
import NewChat from "../../ChatInSocial/src/NewChat";
import GroupProfileChatScreen from "../../GroupProfileChat/src/GroupProfileChat";
import Meetings from "../../MeetingsInSocial/src/Meetings";
import More from "../../MoreInSocial/src/More";
import ProfileChat from "../../ProfileChat/src/ProfileChat";
import * as IMAGE from "./assets";
import ChatDashBoard from "../../ChatInSocial/src/ChatDashBoard";
import UsersSelection from "../../ChatInSocial/src/UsersSelection";
import CreateNewGroup from "../../ChatInSocial/src/CreateNewGroup";

const HomeStack = createBottomTabNavigator(
  {
    Channels: {
      screen: Channels,
      navigationOptions: { header: null, title: "Channels" },
    },
    Chat: { screen: ChatDashBoard, navigationOptions: { header: null, title: "Chat" } },
    Meetings: {
      screen: Meetings,
      navigationOptions: { header: null, title: "Meetings" },
    },
    More: { screen: More, navigationOptions: { title: "More", header: null } },
    ProfileChat: {
      screen: ProfileChat,
      navigationOptions: {
        title: "ProfileChat",
        tabBarVisible: false,
        header: null,
      },
    },
    GroupProfileChatScreen: {
      screen: GroupProfileChatScreen,
      navigationOptions: {
        header: null,
        tabBarVisible: false,
        title: "GroupProfileChatScreen",
      },
    },
    ConversationScreen: {
      screen: ConversationScreen,
      navigationOptions: {
        title: "ConversationScreen",
        tabBarVisible: false,
        header: null,
      },
    },
    UsersSelection: {
      screen: UsersSelection,
      navigationOptions: {
        header: null,
        tabBarVisible: false,
        title: "UsersSelection",
      },
    },
    CreateNewGroup: {
      screen: CreateNewGroup,
      navigationOptions: {
        header: null,
        tabBarVisible: false,
        title: "CreateNewGroup",
      },
    },
  },
  {
    tabBarComponent: (props: any) => {
      const activeTabIndex: number = props.navigation.state.index;
      return (
        <View
          style={{
            height: 80,
            paddingVertical: 25,
            alignItems: "center",
            flexDirection: "row",
            justifyContent: "space-evenly",
            backgroundColor: "#474740",
          }}
        >
          <TabBarComponent
            path={IMAGE.hash}
            text="Channels"
            activeTab={activeTabIndex}
            tabId={0}
            onTabPress={() => props.navigation.navigate("Channels")}
          />
          <TabBarComponent
            testID="TabChatID"
            path={IMAGE.chat}
            text="Chat"
            activeTab={activeTabIndex}
            tabId={1}
            onTabPress={() => props.navigation.navigate("Chat")}
          />
          <TabBarComponent
            path={IMAGE.meeting}
            text="Meetings"
            activeTab={activeTabIndex}
            tabId={2}
            onTabPress={() => props.navigation.navigate("Meetings")}
          />
          <TabBarComponent
            path={IMAGE.more}
            text="More"
            activeTab={activeTabIndex}
            tabId={3}
            onTabPress={() => props.navigation.navigate("More")}
          />
        </View>
      );
    },
  }
);

export default class SocialCommunication extends Component {
  constructor(props: any) {
    super(props);
  }

  render() {
    const screenProps: any = {
      navigationProp: this.props.navigation.state.params.navigationProp,
    };

    return <HomeStack screenProps={screenProps} />;
  }
}
export interface TabBarTypes {
  path: ImageSourcePropType;
  text: string;
  activeTab: number;
  tabId: number;
  onTabPress: () => any;
}
export class TabBarComponent extends Component<TabBarTypes> {
  constructor(props: TabBarTypes) {
    super(props);
  }

  colorTint = () => {
    if (this.props.activeTab == this.props.tabId) {
      return "rgb(255, 255, 255)";
    } else {
      return "rgb(255, 255, 256)";
    }
  };
  render() {
    return (
      <TouchableOpacity
        testID="TabID"
        onPress={this.props.onTabPress}
        style={Styles.tabBarButton}
      >
        <Image
          source={this.props.path}
          resizeMode="contain"
          style={[Styles.tabBarImage, { tintColor: this.colorTint() }]}
        />
        <Text
          style={[
            Styles.tabBarText,
            {
              color: this.colorTint(),
            },
          ]}
        >
          {this.props.text}
        </Text>
      </TouchableOpacity>
    );
  }
}
const Styles = StyleSheet.create({
  tabBarButton: {
    justifyContent: "center",
    alignItems: "center",
  },
  tabBarImage: {
    width: 23,
    height: 23,
  },
  tabBarText: {
    fontWeight: "500",
  },
});
// Customizable Area End
