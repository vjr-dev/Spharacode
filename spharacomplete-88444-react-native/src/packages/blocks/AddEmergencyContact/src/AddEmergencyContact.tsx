// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  Image,
  ImageBackground,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  ScrollView,
  Modal,
  ActivityIndicator
} from "react-native";
import { COLORS } from "framework/src/Globals";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import AddEmergencyContactController, {
  Props,
} from "./AddEmergencyContactController";
import { back1, back2, image_back, Profile } from "./assets";
import { scaledSize } from "../../../framework/src/Utilities";
import { deviceHeight, deviceWidth } from "framework/src/Utilities";
import { displayConfirmAlert } from "../../../components/src/CustomAlert";

export default class AddEmergencyContact extends AddEmergencyContactController {
  constructor(props: Props) {
    super(props);
  }

  getContactType = (type) => {
    if( type === 'emergency_contact'){
      return "Emergency Contact"
    }else if(type === 'friends'){
      return 'Friends'
    }else{
      return 'Family'
    }
  }
  renderItem = (item) => {
    return (
      <View style={styles.itemView}>
        <View style={ styles.itemInnerView}>
          <Image
            source={Profile}
            style={styles.profileImage}
          />
          <View style={styles.textView}>
            <View style={{flex:1}}>
              <Text style={styles.name} numberOfLines={1}>{item.attributes.name}</Text>
              <Text testID={"phoneNumberText"} style={styles.number} numberOfLines={1}>{item.attributes.phone_number}</Text>
              <Text style={styles.contactType} numberOfLines={1}>{this.getContactType(item.attributes.contact_type)} </Text>
            </View>
            {this.state.ListData.length == 1 ? null : (
            <TouchableOpacity
              testID="addEmergencyContactDeleteButton"
              onPress={() => this.deleteItemConfirmation(item)}
              style={styles.deleteButton}
            >
              <MaterialIcon name="close" color={COLORS.black} size={scaledSize(17)} />
            </TouchableOpacity>
          )}
          </View>
        </View>
      </View>
    );
  };
  deleteItemConfirmation(item: any) {
    displayConfirmAlert("Are you sure?","Do you want to Delete Emergency contact?", ()=> this.deleteContacts(item.id))
  }
  render() {
    return (
      <SafeAreaView style={styles.container}>
        <ImageBackground source={back1} style={styles.bgImage}>
          <ImageBackground source={back2} style={styles.bgImage}>
            <View style= {styles.body}>
              <View style={styles.headerView} >
                <TouchableOpacity
                  testID="addEmergencyContactBackButton"
                  style={styles.backButton}
                  onPress={() => this.handleBackButtonClick()}
                >
                  <Image
                    source={image_back}
                    style={styles.backIcon}
                  />
                </TouchableOpacity>
                
                <TouchableOpacity
                  testID="addEmergencyContactOpenAddNewModalButton"
                  onPress={() => this.onAddNewContacts()}
                  style={styles.addNewButton}
                >
                  <Text style={styles.addNewLabel}> ADD NEW </Text>
                </TouchableOpacity>
              </View>
              <Text style={styles.title}> Add Emergency Contacts.</Text>
              <Text style={styles.label}>
                Your emergency contacts will receive your message  when you trigger the panic button.
              </Text>
              <View style={styles.contactView}>
                <ScrollView showsVerticalScrollIndicator={false}>
                  {
                    this.state.ListData.map((item: any)=>{
                      return (this.renderItem(item));
                    })
                  }
                </ScrollView>
              </View>
              <TouchableOpacity
                testID="addEmergencyContactDoneButton"
                onPress={() => {
                  this.onDoneClick();
                }}
                style={styles.doneButton}
              >
                <Text style={{ color: "#fff", fontWeight: "bold" }}>DONE</Text>
              </TouchableOpacity>
            </View>
            <Modal visible={this.state.isLoading} transparent={true}>
              <View style={styles.loaderContainer}>
                <ActivityIndicator
                  animating={true}
                  color="#f07233"
                  hidesWhenStopped={true}
                  size="large"
                />
              </View>
            </Modal>
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1, 
    backgroundColor: "#454545"
  },
  bgImage: {
    height: deviceHeight,
    width: deviceWidth,
    flex: 1,
  },
  body:{
    flex:1,
    padding: scaledSize(10),
  },
  headerView:{
    height: scaledSize(30),
    width: "100%",
    marginVertical: scaledSize(10),
    justifyContent: "space-between",
    alignItems:'center',
    flexDirection: "row",
  },
  backButton:{
    height: scaledSize(30), 
    width: scaledSize(30), 
    justifyContent:'center',
    alignItems:'center'
  },
  backIcon:{
    height: scaledSize(15), 
    width: scaledSize(15), 
    tintColor: "#f07233"
  },
  addNewButton:{
    height: scaledSize(30), 
    paddingHorizontal: scaledSize(10),
    justifyContent:'center',
    alignItems:'center'
  },
  addNewLabel:{
    color: "#f07233", 
    fontSize: scaledSize(13), 
  },
  title:{
    fontSize: scaledSize(20),
    color: "#fff",
  },
  label:{
    fontSize: scaledSize(12),
    color: "#cdcdcd",
    marginTop: scaledSize(10),
  },
  contactView:{
    flex:1,
    paddingVertical: scaledSize(10)
  },
  doneButton:{
    height: scaledSize(40),
    width: "100%",
    marginBottom: scaledSize(15),
    backgroundColor: "#f07233",
    borderRadius: scaledSize(20),
    alignItems: "center",
    justifyContent: "center",
  },
  itemView:{
    alignSelf: "center",
    marginVertical: scaledSize(5),
    alignItems: "center",
  },
  itemInnerView:{
    flexDirection: "row",
    alignItems:'center',
    justifyContent:'center'
  },
  profileImage:{
    height: scaledSize(40),
    width: scaledSize(40),
    marginRight: scaledSize(10),
    borderRadius: scaledSize(20),
  },
  textView:{
    flex:1,
    flexDirection:'row',
    alignItems:'center',
    justifyContent:'space-between',
    paddingBottom: scaledSize(5),
    borderBottomWidth: 0.5,
    borderColor:"#adadad"
  },
  name:{
    color: "#adadad", 
    fontSize: scaledSize(12)
  },
  number:{
    color: "#adadad", 
    fontSize: scaledSize(11)
  },
  contactType:{
    color: "#adadad", 
    fontSize: scaledSize(10)
  },
  deleteButton:{
    height: scaledSize(20),
    width: scaledSize(20),
    marginRight: scaledSize(20),
    marginLeft: scaledSize(10),
    backgroundColor: "#adadad",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: scaledSize(10),
  },
  loaderContainer:{
    flex:1,
    alignItems:'center',
    justifyContent:'center'
  }
});
// Customizable Area End
