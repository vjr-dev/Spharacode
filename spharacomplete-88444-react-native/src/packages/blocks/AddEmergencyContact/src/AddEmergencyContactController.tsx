// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { BlockComponent } from "../../../framework/src/BlockComponent";
import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { GotoHomePage } from "../../../components/src/Navigation/NavigationFunctions";
import { backToLoginConfirmationAlert, displayErrorMessage, displaySuccessMessage } from "../../../components/src/CustomAlert";
import { BackHandler } from "react-native";
import { OnLogOut } from "../../../components/src/Navigation/logout";
export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  ListData: any;
  Token: string;
  isLoading: boolean
}

interface SS {
  id: any;
}

export default class EmergencyContact extends BlockComponent<Props, S, SS> {
  getListId: any = "";
  DeletecontactId: any = "";

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.RestAPIRequestMessage),
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      getName(MessageEnum.RestAPIRequestBodyMessage),
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      getName(MessageEnum.RestAPIRequestMethodMessage),
    ];

    this.state = {
      ListData: [],
      Token: "",
      isLoading: false
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async componentDidMount() {
    this.setState({ isLoading: true });
    let token: any = await AsyncStorage.getItem("Token");
    this.setState({ Token: token });
    this.getContacts();
    this.props.navigation.addListener('focus',()=>{
      this.getContacts();
    })
  }
  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackButtonClick);
  }

  componentWillUnmount() {
      BackHandler.removeEventListener('hardwareBackPress', this.handleBackButtonClick);
  }

  handleBackButtonClick = async () => {
    if(this.props.route.params?.from !== 'EmergencyContact'){
      this.props.navigation.pop();
      return true;
    }else{
      backToLoginConfirmationAlert(async ()=> {
        this.setState({isLoading: true})
        await OnLogOut(this.props)
        this.setState({isLoading: false})
      });
      return true;
    }
  }
  onAddNewContacts = () => {
    this.props.navigation.navigate("EmergencyContact", {from: "AddContact", contactType: this.props.route.params.contactType})
  }
  receiveGetList(responseJson: any) {
    if (responseJson != null) {
      if (responseJson.errors != null) {
        this.setState({ isLoading: false });
        displayErrorMessage(responseJson.errors[0]);
      } else {
        this.setState({ ListData: responseJson.data });
        this.setState({ isLoading: false });
      }
    }
  }
  receiveDeletecontact(responseJson: any) {
    if (responseJson != null) {
      if (responseJson.errors != null) {
        this.setState({ isLoading: false });
        displayErrorMessage(responseJson.errors[0]);
      } else {
        this.getContacts();
        displaySuccessMessage("Contact delete successfully.")
      }
    }
  }
  async receive(from: string, message: Message) {
    runEngine.debugLog("Message Recived", message);

    if (message.id === getName(MessageEnum.RestAPIResponceMessage)) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      let responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      if (apiRequestCallId === this.getListId) {
        this.receiveGetList(responseJson);
      }
      if (apiRequestCallId === this.DeletecontactId) {
        this.receiveDeletecontact(responseJson);
      }
    }
  }

  getContacts() {
    this.setState({ isLoading: true });
    const header = {
      "Content-Type": configJSON.GetcontactApiContentType,
      token: this.state.Token,
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.getListId = requestMessage.messageId;

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.GetcontactAPiEndPoint
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.GetcontactAPiMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);

    return true;
  }


  async onDoneClick() {
    if(this.props.route.params?.from !== 'EmergencyContact'){
      this.props.navigation.pop();
    }else{
      GotoHomePage(this.props);
    }
  }

  deleteContacts(id: string) {
    this.setState({ isLoading: true });
    const header = {
      "Content-Type": configJSON.DeletecontactApiContentType,
      token: this.state.Token,
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.DeletecontactId = requestMessage.messageId;

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      `bx_block_contact_us/contacts/${id}`
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.DeletecontactAPiMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);

    return true;
  }
}
// Customizable Area End
