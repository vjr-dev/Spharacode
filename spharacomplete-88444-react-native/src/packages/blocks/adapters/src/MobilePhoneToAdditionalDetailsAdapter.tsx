import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
