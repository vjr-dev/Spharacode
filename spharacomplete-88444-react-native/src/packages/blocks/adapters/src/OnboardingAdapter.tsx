import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
