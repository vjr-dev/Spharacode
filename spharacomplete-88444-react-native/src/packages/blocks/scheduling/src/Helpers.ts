/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});