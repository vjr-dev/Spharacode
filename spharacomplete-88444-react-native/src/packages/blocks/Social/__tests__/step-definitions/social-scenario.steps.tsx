//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";
import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import SocialCommunication, { TabBarComponent } from "../../src/SocialCommunication";

let navigationStateIndex;

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    props: jest.fn(),
    dispatch: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),

    state: {
      params: {},
      index: navigationStateIndex,
    },
  },
  id: "Social",
};

const screenProps1 = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    props: jest.fn(),
    dispatch: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),

    state: {
      params: {},
      index: navigationStateIndex,
    },
  },
  id: "TabBar",
};

const feature = loadFeature("./__tests__/features/social-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to SocialCommunication", ({ given, when, then }) => {
    let socialWrapper: ShallowWrapper;
  
    given("I am a User attempting to navigate SocialCommunication", () => {
      socialWrapper = shallow(<SocialCommunication {...screenProps} />);
      expect(socialWrapper).toBeTruthy();
      expect(socialWrapper).toMatchSnapshot();
    });

    when("I navigate to the SocialCommunication", () => {
      expect(socialWrapper).toBeTruthy();
    });
  });

  test("User navigates to TabBarComponent", ({ given, when, then }) => {
    let tabBarWrapper: ShallowWrapper;

    given("I am a User attempting to navigate TabBarComponent", () => {
      tabBarWrapper = shallow(<TabBarComponent {...screenProps1} />);
      expect(tabBarWrapper).toBeTruthy();
      expect(tabBarWrapper).toMatchSnapshot();
      let buttonComponent1 = tabBarWrapper.findWhere(
        (node) => node.prop("testID") === "TabID"
      );
      buttonComponent1.simulate("press");
    });

    when("I navigate to the TabBarComponent", (index) => {
      navigationStateIndex = index;
    });
  });
});
