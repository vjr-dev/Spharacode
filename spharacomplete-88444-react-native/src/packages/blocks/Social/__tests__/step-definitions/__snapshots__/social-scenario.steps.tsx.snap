// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social User navigates to SocialCommunication 1`] = `
<NavigationContainer
  screenProps={
    Object {
      "navigationProp": undefined,
    }
  }
/>
`;

exports[`Social User navigates to TabBarComponent 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="TabID"
>
  <Image
    resizeMode="contain"
    style={
      Array [
        Object {
          "height": 23,
          "width": 23,
        },
        Object {
          "tintColor": "rgb(255, 255, 255)",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontWeight": "500",
        },
        Object {
          "color": "rgb(255, 255, 255)",
        },
      ]
    }
  />
</TouchableOpacity>
`;
